nodeadm/internal/util/sysinfo.go

```diff
@@ -0,0 +1,286 @@
+package util
+
+import (
+	"fmt"
+	"go.uber.org/zap"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	cpuDirRegExp = regexp.MustCompile(`/cpu(\d+)`)
+	nodeDir      = "/sys/devices/system/node"
+	cpusPath     = "/sys/devices/system/cpu"
+)
+
+const (
+	sysFsCPUTopology = "/topology"
+	cpuDirPattern    = "cpu*[0-9]"
+	nodeDirPattern   = "node*[0-9]"
+
+	coreIDFilePath    = sysFsCPUTopology + "/core_id"
+	packageIDFilePath = sysFsCPUTopology + "/physical_package_id"
+)
+
+type Core struct {
+	Id       int   `json:"core_id"`
+	Threads  []int `json:"thread_ids"`
+	SocketID int   `json:"socket_id"`
+}
+
+func init() {
```

Comment:
need this as the mock sys files cannot be copied into the /sys during docker image build, this might be a hack, let me know if there is better way to set up this mock

Reply:
Do you mean they get wiped on boot or is it a perm issue? Sorry I haven't tested myself yet

Would a bash helper that u call in the test cases to copy them into the dir work?